### DISCUSSION TITLE
Provide a way to quickly and easily check that the password is valid

### DISCUSSION THREAD

**Discussion by ra-dave** [rikyoz/bit7z#145] [Ideas]
I see that the test() method apparently validates that the password is correct, which is great, but for a large archive file it takes a while for test() to complete.  I would like the ability to quickly test whether the password the user supplied is valid.  I know I can do this by trying to extract a single file but that is clunky.

BTW, when test() throws an exception due to a bad password the exception message is "Unknown Error: Unspecified error".  Perhaps a more descriptive message could be used?

**Comment by rikyoz** [comment]
Hi!

> I see that the test() method apparently validates that the password is correct, which is great, but for a large archive file it takes a while for test() to complete. I would like the ability to quickly test whether the password the user supplied is valid. I know I can do this by trying to extract a single file but that is clunky.

Unfortunately, 7-zip doesn't provide a direct way to check if the password is correct (as far as I know, at least).

The main problem I see with a method that verifies a password's validity is that, in the most general case possible, an archive may contain several files encrypted with different passwords or files that are encrypted and others that are not. So this method would not differ much from the current test() method.

A test function that tests only a specific file inside the archive might speed up your use case, though. But it will leave to the user to know which file to test.

> BTW, when test() throws an exception due to a bad password the exception message is "Unknown Error: Unspecified error". Perhaps a more descriptive message could be used?

Yeah, I'll definitely work on improving the exception messages before releasing the stable v4.
Sadly, the 7-zip DLL is a bit inconsistent in the error reporting; for example, it reports a wrong password error when the archive format is Zip or Rar5 but not when it's 7z.
So I'll need to find some workaround for that!

Anyway, thank you for the suggestions! 🙏
